x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/DeleteConnectorSyncJobAction.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector.syncjob.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+
+public class DeleteConnectorSyncJobAction extends ActionType<AcknowledgedResponse> {
+
+    public static final DeleteConnectorSyncJobAction INSTANCE = new DeleteConnectorSyncJobAction();
+    public static final String NAME = "cluster:admin/xpack/connector/sync_job/delete";
+
+    private DeleteConnectorSyncJobAction() {
+        super(NAME, AcknowledgedResponse::readFrom);
+    }
+
+    public static class Request extends ActionRequest implements ToXContentObject {
+        public static final String EMPTY_CONNECTOR_SYNC_JOB_ID_ERROR_MESSAGE =
+            "[connector_sync_job_id] of the connector sync job cannot be null or empty.";
+        public static final ParseField CONNECTOR_SYNC_JOB_ID_FIELD = new ParseField("connector_sync_job_id");
```

Comment:
nit: this is likely to be reused by other requests, should we put it to `SyncJob` class?  Can be addressed in the next PR

Reply:
Yeah I think we could've a `Connectors` class maybe similar to `EnterpriseSearch`, which holds some of these constants. Definitely open to discuss.